Zotero/Scenes/General/Views/ContainerViewController.swift

```diff
@@ -35,30 +35,120 @@ class ContainerViewController: UIViewController {
+
+                button.frame = view.bounds
+                button.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+                view.insertSubview(button, at: 0)
+            }
+
+            func setupKeyboardObserving() {
+                NotificationCenter.default
+                    .keyboardWillShow
+                    .observe(on: MainScheduler.instance)
+                    .subscribe(onNext: { notification in
+                        guard let data = notification.keyboardData else { return }
+                        moveToKeyboard(data, willShow: true)
+                    })
+                    .disposed(by: disposeBag)
+
+                NotificationCenter.default
+                    .keyboardWillHide
+                    .observe(on: MainScheduler.instance)
+                    .subscribe(onNext: { notification in
+                        guard let data = notification.keyboardData else { return }
+                        moveToKeyboard(data, willShow: false)
+                    })
+                    .disposed(by: disposeBag)
+
+                func moveToKeyboard(_ data: KeyboardData, willShow: Bool) {
+                    guard let containerCenterY else { return }
+
+                    containerCenterY.constant = willShow ? data.endFrame.height / 2 : 0
+
+                    guard isViewLoaded else { return }
+
+                    UIView.animate(withDuration: 0.2) {
+                        self.view.layoutIfNeeded()
+                    }
+                }
+            }
+        }
+
+        func setupForPhone() {
+            NSLayoutConstraint.activate([
+                rootViewController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                rootViewController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+                rootViewController.view.topAnchor.constraint(equalTo: view.topAnchor),
+                rootViewController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+            ])
         }
     }
 
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
-        self.didAppear = true
+        didAppear = true
     }
 
     override func preferredContentSizeDidChange(forChildContentContainer container: UIContentContainer) {
         super.preferredContentSizeDidChange(forChildContentContainer: container)
 
-        guard let height = self.containerHeight, let width = self.containerWidth else { return }
+        guard let containerHeight, let containerWidth else { return }
 
-        height.constant = container.preferredContentSize.height
-        width.constant = container.preferredContentSize.width
+        containerHeight.constant = container.preferredContentSize.height
+        containerWidth.constant = container.preferredContentSize.width
 
-        guard self.didAppear else { return }
+        guard didAppear else { return }
 
         UIView.animate(withDuration: 0.2) {
             self.view.layoutIfNeeded()
```

Comment:
Since you removed `self`s everywhere, you can remove this one too.

Reply:
Check https://github.com/zotero/zotero-ios/pull/827/commits/72089cea9a35bc6efa387ee9896b4b38a16a659b